### PR TITLE
Added missing C++ enum values

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation
Fix warnings for missing enum values in switch statements

### CODE REVIEW

File: pulsar-client-cpp/lib/ClientConnection.cc [hunk 1]
```diff
@@ -97,6 +97,12 @@ static Result getResult(ServerError serverError) {
 
         case UnsupportedVersionError:
             return ResultUnsupportedVersionError;
+
+        case TooManyRequests:
+            return ResultTooManyLookupRequestException;
```
Comment (jai1) [inline pulsar-client-cpp/lib/ClientConnection.cc]: ResultTooManyRequestException - in case we want to ratelimit something else in the future

**Comment by jai1** [comment]
retest this please

File: pulsar-client-cpp/lib/ClientConnection.cc (hunk 1 above)
Comment (merlimat) [reply pulsar-client-cpp/lib/ClientConnection.cc]: The enum value was already defined in `Result`

**Comment by jai1** [comment]
retest this please

**Comment by jai1** [comment]
retest this please

**Pull request merged**